### ISSUE TITLE
Epic Games Store games list includes Unreal Engine library assets/plugins

### ISSUE THREAD

**Issue by Pzenarch** [lutris/lutris#5692]
### Bug description

I have a personal Epic Games account which I use for both playing games and tinkering with Unreal Engine.  When I link this account to Lutris, select 'Epic Games Store' from the sources list, and disable the 'Installed games only' filter, the list includes both the games in my games library, and the Unreal Engine assets/plugins that are in my "Fab" library.

### How to Reproduce

1. Have an Epic Games Store account which has been used for Unreal Engine development and includes some assets (including free ones) from the (Fab) Library.
2. Link this account to Lutris
3. Note that *some* of your (Fab) Library assets/plugins may be listed as if they were games, with some of them being listed multiple times.

** Not all (Fab) Library assets appear in the games list, but I've included below a non-exhaustive list of some which are:
  - Arch Vis Polish Pack #2 - https://www.fab.com/listings/3f15cf5f-0ea0-45d9-896b-ec3c5bf6cb9d
  - Backtrace Crash Reporting Plugin - https://www.fab.com/listings/fd0007dd-d1c0-4f2c-9928-8e27fe62e8be
  - Blueprints - https://www.fab.com/listings/720af3f1-cd24-40a3-a881-ee695a7c9779
  - Epic Zen Garden - https://www.fab.com/listings/27463ce7-56e0-43c7-ad39-d853da0461de

### Expected behavior

I'd suggest that Unreal Engine (Fab) library assets/plugins probably shouldn't be listed in the games list ... I don't see any value in having them there, as even if we want to install them, they should be associated with a specific version of the engine, something that the Lutris client needn't have any knowledge of.

### Log output

```shell
INFO     2024-10-31 14:44:14,868 [application.do_command_line:479]:Starting Lutris 0.5.17
DEBUG    2024-10-31 14:44:14,869 [system._execute:109]:Executing lspci
INFO     2024-10-31 14:44:14,917 [startup.run_all_checks:120]:Advanced Micro Devices, Inc. [AMD/ATI] Navi 31 [Radeon RX 7900 XT/7900 XTX/7900 GRE/7900M] (1002:744c 1462:5200 amdgpu) Driver 24.2.6
ERROR    2024-10-31 14:44:14,918 [startup.check_libs:62]:i386 libgnutls.so.30 missing (needed by gnutls)
DEBUG    2024-10-31 14:44:15,056 [path_cache._update_missing_games:114]:Checking for missing games
DEBUG    2024-10-31 14:44:15,072 [api.check_stale_runtime_versions:52]:Modified at Thu 31 Oct 2024 11:08:45 , will update after Thu 31 Oct 2024 17:08:45 
DEBUG    2024-10-31 14:44:15,076 [lutriswindow.create_runtime_updater_cb:1244]:Runtime up to date
DEBUG    2024-10-31 14:44:17,710 [base.wipe_game_cache:441]:Deleting egs cache /home/fueda/.cache/lutris/egs-library.json
DEBUG    2024-10-31 14:44:17,710 [base.wipe_game_cache:175]:Deleting games from service-games for egs
INFO     2024-10-31 14:46:35,693 [egs.add_installed_games:341]:EGS detected in /home/fueda/Games/epic-games-store
WARNING  2024-10-31 14:46:35,693 [egs_launcher.iter_manifests:18]:No valid path for EGS games manifests in /home/fueda/Games/epic-games-store/drive_c/ProgramData/Epic/EpicGamesLauncher/Data/Manifests
DEBUG    2024-10-31 14:46:35,693 [egs.add_installed_games:352]:All EGS games imported
```

### System Information

```shell
Vulkan support: YES
Esync support: YES
Fsync support: YES
Wine installed: YES
Gamescope: YES
Mangohud: YES
Gamemode: YES
Steam: YES
In Flatpak: NO
[System]
OS: CachyOS Linux rolling n/a
Arch: x86_64
Kernel: 6.11.5-3-cachyos
Desktop: GNOME
Display Server: wayland
[CPU]
Vendor: AuthenticAMD
Model: AMD Ryzen 7 7800X3D 8-Core Processor
Physical cores: 8
Logical cores: 16
[Memory]
RAM: 62.3 GB
Swap: 62.3 GB
[Graphics]
Vendor: AMD
OpenGL Renderer: AMD Radeon RX 7900 XTX (radeonsi, navi31, LLVM 18.1.8, DRM 3.59, 6.11.5-3-cachyos)
OpenGL Version: 4.6 (Compatibility Profile) Mesa 24.2.6-cachyos1.2
OpenGL Core: 4.6 (Core Profile) Mesa 24.2.6-cachyos1.2
OpenGL ES: OpenGL ES 3.2 Mesa 24.2.6-cachyos1.2
Vulkan Version: 1.3.295
Vulkan Drivers: AMD Radeon RX 7900 XTX (RADV NAVI31) (1.3.289)
```

### Media (optional)

![image](https://github.com/user-attachments/assets/c58121fe-4aed-4d15-b5b4-50ff50f22682)

**Comment by Mart-y-McFly** [comment]
Can confirm the bug. Freshly installed Debian Trixie and Lutris 0.5.18.

**Comment by esuriddick** [comment]
I can confirm the same. Freshly installed Linux Mint 22.2. and Lutris 0.5.14.

**Comment by legluondunet** [comment]
Is this issue still present in latest Lutris version 0.5.22?
Without answer, this ticket will be closed in 7 days.

**Comment by Pzenarch** [comment]
Yes.  It's possible some details have changed since the original ticket was raised, but I can still see UE FAB assets listed in my Epic games library under 0.5.22, and no obvious deviations from the ticket as originally raised.

**Comment by Mart-y-McFly** [comment]
Yes, the UE assets are still present.

**Comment by strycore** [comment]
Lutris 🤝 Heroic 🤝 Playtron
   All having the same bug

**Comment by SakuraJensen** [comment]
> Is this issue still present in latest Lutris version 0.5.22? Without answer, this ticket will be closed in 7 days.

~~Can confirm this issue still exists.  I'd be happy to test fixes if needed.~~


Update:  I think I fixed the issue.  I diffed the sqlite entries between known games and known assets and found that there was a difference between the entitlementType of AUDIENCE and EXECUTABLE.  But, I used AI to help write the code quickly so I'm going to attach the changes and my findings here so you can implement it properly and safely. I still haven't checked to make sure none of my actual games were removed but 341 games sounds about right.

Updated again: I have since tested and all of my games are still there, so this seems to be a viable fix.  I don't see anything inherently screaming at me at would be bad about the code either.  

@legluondunet If it's okay with you I could submit a PR.  No problem if you're already working on a different solution though.

[egs-fab-filter-tested.patch](https://github.com/user-attachments/files/28719864/egs-fab-filter-tested.patch)
[lutris-epic-samples.tar.gz](https://github.com/user-attachments/files/28719631/lutris-epic-samples.tar.gz)
[lutris-epic-sample-findings.md](https://github.com/user-attachments/files/28720437/lutris-epic-sample-findings.md)

<img width="1843" height="1202" alt="Image" src="https://github.com/user-attachments/assets/543a01e1-b655-4fea-bbfe-0bf1935c2f6f" />

**Comment by legluondunet** [comment]
Hello @SakuraJensen , if you think you resolved a Lutris issue, don't hesitate to submit a PR.
